pkg/storage/system_intake_test.go

```diff
@@ -998,3 +1001,39 @@ func (s StoreTestSuite) TestUpdateSystemIntakeLinkedCedarSystem() {
 		s.False(updatedIntake.CedarSystemID.Valid)
 	})
 }
+
+func (s StoreTestSuite) TestFetchRelatedSystemIntakes() {
+	ctx := context.Background()
+
+	s.Run("fetch related system intakes", func() {
+		cedarSystemID := null.StringFrom("555-55-6")
+		contractNumber := null.StringFrom("444-55-5")
+
+		intake := testhelpers.NewSystemIntake()
+		relatedIntake1 := testhelpers.NewSystemIntake()
+		relatedIntake2 := testhelpers.NewSystemIntake()
+		relatedIntake3 := testhelpers.NewSystemIntake()
+
+		intake.CedarSystemID = cedarSystemID
+		intake.ContractNumber = contractNumber
+
+		relatedIntake1.CedarSystemID = cedarSystemID
+		relatedIntake2.CedarSystemID = cedarSystemID
+		relatedIntake3.ContractNumber = contractNumber
+
+		tx := s.db.MustBegin()
+		intakesToCreate := []models.SystemIntake{intake, relatedIntake1, relatedIntake2, relatedIntake3}
+		for i := range intakesToCreate {
+			_, err := tx.NamedExec(insertIntakeWithCedarSystemAndContractSQL, &intakesToCreate[i])
+			s.NoError(err)
+		}
+
+		err := tx.Commit()
+		s.NoError(err)
+
+		relatedIntakes, err := s.store.FetchRelatedSystemIntakes(ctx, intake.ID)
+		s.NoError(err)
+
+		s.Equal(len(relatedIntakes), 3)
```

Comment:
I'd modify this test to check the return values (perhaps just the IDs returned), this'll help validate that the logic is working as intended

Reply:
Seconded.